assets/js/base/components/block-error-boundary/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import BlockError from './block-error';
+import './style.scss';
+
+class BlockErrorBoundary extends Component {
```

Comment:
You could add some more utility to this error boundary if:

- it accepts a prop for whether to display the actual caught error message (You can get that via the `componentDidCatch` lifecycle method).
- you always `console.warn` or `console.error` the error message  if development mode (`SCRIPT_DEBUG` is true) is detected (and don't if in production  mode).
- accepts props for passing through to the underlying `BlockError` component (as per my comments on that component)

Reply:
Good ideas! Implemented in 70534f4eeface5651eb6128bf4889fd9024543df.

> - you always `console.warn` or `console.error` the error message  if development mode (`SCRIPT_DEBUG` is true) is detected (and don't if in production  mode).

I searched around but couldn't find a way to do that, according to [this comment](https://github.com/facebook/react/issues/15069#issuecomment-472012103) it doesn't seem to be possible?

Reply:
🤔 Hmm [this example](https://codepen.io/gaearon/pen/wqvxGa?editors=0010) doesn't seem to automatically log to the console.  Since the example is using production react I'm guessing the console is automatically logged to in development builds (something you could try?).  If true, then you shouldn't need to worry about `SCRIPT_DEBUG` (we'll get console logging for free in dev environments!)

Also from the example it looks like `comopnentDidCatch` is used instead of `getDerivedErrorFromState`.  So you could use both here (the extra info (including the component stack) would likely be useful in certain contexts as well.

Reply:
I was wrong, the example *does* console.log :( . So the question then is whether it matters if the console has the error?

Reply:
Sorry for the waffling, turns out the example was loading development react (I thought it was loading production).  I switched the example to load production react and _no_ console logging.  So I think we can use this approach.

Reply:
I'm still seeing the error log when using production React:
![image](https://user-images.githubusercontent.com/3616980/68879579-f1bcbe00-0709-11ea-8379-95c833efc5bc.png)

Did you do anything else besides changing the loading scripts?

Reply:
Hmm, but interestingly, if you look at the development vs production results, the _exposed_ info in the console is much different.

Why I thought there was nothing put in the console is because I forgot I commented out `componentDidCatch`.  `getDerivedStateFromError` will console.log in development mode but not in production.

So, I say all that to say that it looks like you CAN safely use `componentDidCatch` because even though there will be a console error in production mode, it's not exposing anything (other than there was an error, and the path of the error).  Whereas in development mode there's more useful information in the error stack (which component the error happened in etc)